catch/include/resource_guards.hh

```diff
@@ -29,10 +29,30 @@ enum class LinearAllocs {
   hipHostMalloc,
   hipMalloc,
   hipMallocManaged,
+  noAlloc
 };
 
+inline std::string to_string(const LinearAllocs allocation_type) {
+  switch (allocation_type) {
+    case LinearAllocs::malloc:
+      return "host pageable";
+    case LinearAllocs::mallocAndRegister:
+      return "registered";
+    case LinearAllocs::hipHostMalloc:
+      return "host pinned";
+    case LinearAllocs::hipMalloc:
+      return "device malloc";
+    case LinearAllocs::hipMallocManaged:
+      return "managed";
+    default:
+      return "unknown alloc type";
```

Comment:
I'm not sure these strings are informative enough and/or translate easily to API usage.